components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/action/PreIssueAccessTokenResponseProcessor.java

```diff
@@ -154,14 +163,56 @@ private void logOperationExecutionResults(ActionType actionType,
     }
 
     @Override
-    public ActionExecutionStatus processErrorResponse(Map<String, Object> map, Event event,
+    public ActionExecutionStatus<Failure> processFailureResponse(Map<String, Object> eventContext,
+                                                        Event actionEvent,
+                                                        ActionInvocationFailureResponse failureResponse) throws
+            ActionExecutionResponseProcessorException {
+
+        processServerError(failureResponse.getFailureReason());
+        return new FailedStatus(new Failure(failureResponse.getFailureReason(),
+                failureResponse.getFailureDescription()));
+    }
+
+    private void processServerError(String errorCode)
+            throws ActionExecutionResponseProcessorException {
```

Comment:
Can take this to one line it seems

Reply:
This is not possible due to checkstyle configuration.